custom-routes-vis/src/main/webapp/script.js

```diff
@@ -335,15 +335,56 @@ function selectRouteDisplayDetails(
   let routeInfoElement = document.getElementById('route-info');
   routeInfoElement.innerText = 'Selected Route Info:' +
       '\nDuration: ' + formatDuration(totalDurationSec) +
-      '\nDistance: ' + formatDistance(totalDistanceMeters);
+      '\nDistance: ' + formatDistance(totalDistanceMeters) + '\n';
 
   if (routeToken !== '') {
-    routeInfoElement.innerText += '\nRoute Token: check the developer console';
+    let copyBtn = document.createElement('button');
+    copyBtn.innerText = 'Copy Route Token';
+    copyBtn.addEventListener('click', function() {
+      copyTokenToClipBoard(routeToken);
+    });
+    routeInfoElement.appendChild(copyBtn);
+
     console.log('Route token:', routeToken);
     updateDeepLinkingUrl(routeToken);
   }
 }
 
+/** Copies the given route token to the clipboard. */
+function copyTokenToClipBoard(token) {
+  // Note: navigator.clipboard is only supported by pages served over HTTPS.
+  if (!navigator.clipboard) {
+    fallbackCopyTokenToClipboard(token);
+    return;
+  }
+  navigator.clipboard.writeText(token).then(function() {
+    alert('Successfully copied to clipboard!');
+  }, function(err) {
+    alert('Could not copy to clipboard.');
+    console.error('Async: Could not copy token', err);
+  });
+}
+
+/** Creates a textarea for the given token and copies it to clipboard. */
+function fallbackCopyTokenToClipboard(token) {
+  var textArea = document.createElement('textarea');
+  textArea.value = token;
+  document.body.appendChild(textArea);
+  textArea.focus();
+  textArea.select();
+
+  try {
+    var successful = document.execCommand('copy');
```

Comment:
Please keep variable names relevant and descriptive.
```suggestion
    var copyCommandStatus = document.execCommand('copy');
```

Reply:
ok